### PR TITLE
3 define user domain

### LINKED ISSUE #3: Define user domain
- model
- enum
- dto
- repository

### PR DESCRIPTION
**Pull request by manas-chandan**
#3 
- [x] model class add
- [x] enum add
- [x] dto class add
- [x] repository interface add

### CODE REVIEW

File: backend/src/main/java/com/lms/backend/model/User.java
Comment (smruti-banty) [inline backend/src/main/java/com/lms/backend/model/User.java:1]: Typo refferenceNumber.

**Review by smruti-banty — CHANGES_REQUESTED** [review]
change accordingly

**Review by smruti-banty — APPROVED** [review]
LGTM.

**Review by rashmi-little — APPROVED** [review]
looks good.

**Pull request merged**